packages/proxy/src/handler.ts

```diff
@@ -66,40 +63,73 @@ function isRedirect(
 
 export async function handler(event: CloudFrontRequestEvent) {
   const { request } = event.Records[0].cf;
+  const domainName = request.origin!.s3!.customHeaders['x-env-domain-name'][0].value;
   const configEndpoint = request.origin!.s3!.customHeaders[
     'x-env-config-endpoint'
   ][0].value;
-  const apiEndpoint = request.origin!.s3!.customHeaders['x-env-api-endpoint'][0]
+  let apiEndpoint = request.origin!.s3!.customHeaders['x-env-api-endpoint'][0]
     .value;
+
   let headers: Record<string, string> = {};
+  let deploymentIdentifier = undefined;
+
+  // We need to re-fetch the proxy config for every request, because it could be
+  // made to a different deployment than the previous request.
+  let proxyConfig = undefined;
+
+  const hostHeader = event.Records[0].cf.request.headers.host[0].value;
```

Comment:
TS compiler option to require checked index access would boost code quality confidence here. Then the `[0]` access would be seen as a nullable type and that would require either doing `?.` access or `!` or adding validation.

Reply:
Fixed.